tests/utils/chain.go

```diff
@@ -7,132 +7,123 @@ import (
-		return common.Hash{}, err
+		return hash, fmt.Errorf("cannot post RPC: %w", err)
 	}
-	return common.MustHexToHash(hash), nil
+
+	return hexStringBodyToHash(respBody)
 }
 
 // GetFinalizedHead calls the endpoint chain_getFinalizedHead to get the latest finalised head
-func GetFinalizedHead(ctx context.Context, t *testing.T, rpcPort string) common.Hash {
+func GetFinalizedHead(ctx context.Context, rpcPort string) (
+	hash common.Hash, err error) {
 	endpoint := NewEndpoint(rpcPort)
 	method := ChainGetFinalizedHead
 	const params = "[]"
 	respBody, err := PostRPC(ctx, endpoint, method, params)
-	require.NoError(t, err)
+	if err != nil {
+		return hash, fmt.Errorf("cannot post RPC: %w", err)
+	}
 
-	var hash string
-	err = DecodeRPC(respBody, &hash)
-	require.NoError(t, err)
-	return common.MustHexToHash(hash)
+	return hexStringBodyToHash(respBody)
 }
 
 // GetFinalizedHeadByRound calls the endpoint chain_getFinalizedHeadByRound to get the finalised head at a given round
 // TODO: add setID, hard-coded at 1 for now
-func GetFinalizedHeadByRound(ctx context.Context, t *testing.T, rpcPort string, round uint64) (common.Hash, error) {
+func GetFinalizedHeadByRound(ctx context.Context, rpcPort string, round uint64) (
+	hash common.Hash, err error) {
 	p := strconv.Itoa(int(round))
 	endpoint := NewEndpoint(rpcPort)
 	method := ChainGetFinalizedHeadByRound
 	params := "[" + p + ",1]"
 	respBody, err := PostRPC(ctx, endpoint, method, params)
-	require.NoError(t, err)
-
-	var hash string
-	err = DecodeRPC(respBody, &hash)
 	if err != nil {
-		return common.Hash{}, err
+		return hash, fmt.Errorf("cannot post RPC: %w", err)
 	}
 
-	return common.MustHexToHash(hash), nil
+	return hexStringBodyToHash(respBody)
 }
 
 // GetBlock calls the endpoint chain_getBlock
-func GetBlock(ctx context.Context, t *testing.T, rpcPort string, hash common.Hash) *types.Block {
+func GetBlock(ctx context.Context, rpcPort string, hash common.Hash) (
+	block *types.Block, err error) {
 	endpoint := NewEndpoint(rpcPort)
 	method := ChainGetBlock
 	params := fmt.Sprintf(`["%s"]`, hash)
 	respBody, err := PostRPC(ctx, endpoint, method, params)
-	require.NoError(t, err)
-
-	block := new(modules.ChainBlockResponse)
-	err = DecodeRPC(respBody, block)
 	if err != nil {
-		return nil
+		return nil, fmt.Errorf("cannot post RPC: %w", err)
 	}
 
-	header := block.Block.Header
-
-	parentHash, err := common.HexToHash(header.ParentHash)
-	require.NoError(t, err)
-
-	nb, err := common.HexToBytes(header.Number)
-	require.NoError(t, err)
-	number := common.BytesToUint(nb)
+	rpcBlock := new(modules.ChainBlockResponse)
+	err = DecodeRPC(respBody, rpcBlock)
+	if err != nil {
+		return nil, fmt.Errorf("cannot decode RPC response body: %w", err)
+	}
 
-	stateRoot, err := common.HexToHash(header.StateRoot)
-	require.NoError(t, err)
+	rpcHeader := rpcBlock.Block.Header
+	header, err := headerResponseToHeader(rpcHeader)
+	if err != nil {
+		return nil, fmt.Errorf("malformed RPC header: %w", err)
```

Comment:
I think the variable `header` is not from the RPC but from the block returned by the RPC call, so the message error could be `malformed block header: %w`